code/modules/spells/spell_types/barnyard.dm

```diff
@@ -12,7 +12,7 @@
 	range = 7
 	cooldown_min = 30
 	selection_type = "range"
-	var/list/compatible_mobs = list(/mob/living/carbon/human, /mob/living/carbon/monkey)
+	var/list/compatible_mobs_typecache = list(/mob/living/carbon/human, /mob/living/carbon/monkey)
```

Comment:
I believe you will need to do this, in order for the `is_type_in_typecache` check to work correctly:
```suggestion
	var/list/compatible_mobs_typecache = typecacheof(/mob/living/carbon/human, /mob/living/carbon/monkey)
```
I suggest testing though.